api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java

```diff
@@ -2085,9 +2088,6 @@ private Criteria createSearchConceptMapCriteria(String code, String sourceName,
 			criteria.addOrder(Order.asc("concept.retired"));
 		}
 
-		// we only want distinct concepts
-		criteria.setResultTransformer(DistinctRootEntityResultTransformer.INSTANCE);
```

Comment:
Did you intentionally remove this?

Reply:
Yes, but interested in your thoughts.  I think this was a holdover from an earlier time.  The root entity here is a ConceptMap, not a Concept.  I think the assumption was that this was leading the method to return distinct Concepts, but this was not actually the case.  This just resulted in distinct ConceptMaps...

Reply:
I would be cautious about removing this and do so only when am very sure that it does not have unintended side effects. At the very least, i would remove it from a commit different for that of fixing duplicate concept ids.

Reply:
OK, I added it back in @dkayiwa .  I don't think it has any impact, so it's not really a problem (though I think it is likely misleading, assuming I'm understanding it correctly).  Unit test I added passes either way...

Reply:
Am fine removing it from a separate commit.